### PR TITLE
Initialize basic mood tracker app

### PR DESCRIPTION
**Pull request by Steven-Pinkerton**
The user of this can edit the moods.jsonl file contained within this application to add moods to be displayed on the web-page. When editing this file please refer to the following example for how to add a mood:

{"mood": "Good", "when": "2022-10-09 Mon 11:56"}

The following moods are currently supported: Good, Bad, Great, Neutral, Excellent.

### CODE REVIEW

**Comment by srid** [comment]
> However it does compile.

The CI still fails. Can you fix it?

> Scaffolding

What scaffolding? If someone other than Srid visits this PR, can they understand what this PR does without looking at the diff?

---

I just clarified this in https://progression.haskell.page/Topics/PR-guidelines.html (screenshot of new text below):

image

File: tests/Spec.hs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module Spec where
+
+import Data.Maybe (fromJust)
+import MoodTracker.Parser (Mood (Bad), MoodEntry (MoodEntry), moodParse, moodWhat, moodWhen, myParseTimeM)
+import Test.Hspec (describe, hspec, it, shouldBe)
+
+main :: IO ()
+main = do
+  file <- readFileLBS "tests/test.jsonl"
```
Comment (srid) [inline tests/Spec.hs]: Do you really need to read from an external file, or can this be specified inline in Haskell?

File: tests/Spec.hs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  hspec $ do
+    describe "Main" $ do
+      it "moodParse works" $ do
+        moodParse file
+          `shouldBe` Right
+            [MoodEntry {moodWhen = fromJust $ myParseTimeM "2022-10-09 Mon 11:51", moodWhat = Bad}]
```
Comment (srid) [inline tests/Spec.hs]: Is your test as exhaustive as possible? 

If your parser had a bug where it failed to parse multiple lines (say, it parses the first line but not the second line) will your test catch it?

**Review by srid — CHANGES_REQUESTED** [review]
Reviewed just the test part.

File: tests/Spec.hs (hunk 1 above)
Comment (Steven-Pinkerton) [reply tests/Spec.hs]: You can however it's easier to add more lines to test to an external file. Then it is to write a string inline with multiple escape characters. It makes it easier to read.

File: tests/Spec.hs (hunk 2 above)
Comment (Steven-Pinkerton) [reply tests/Spec.hs]: I have now added multiple lines to the test.

**Comment by srid** [comment]
Obviously, you should fix the error instead of removing the check.

**Review by srid — APPROVED** [review]

**Pull request merged**